scriptlets/refresh_zapper_if_needed

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+# Update the Zapper snap (and firmware) if needed
+#
+# Description:
+#
+# If a Zapper is part of the setup, refresh the snap
+# to the target channel (default is `beta`), upgrade firmware
+# and wait until the service is stable. 
+
+usage() {
+    echo "Usage: $(basename ${BASH_SOURCE[0]}) [--channel]"
+}
+
+check_for_zapper () {
+    (
+        trap EXIT
+        source "$(dirname ${BASH_SOURCE[0]})/defs/check_for_zapper_ip" > /dev/null
+    )
+}
+
+wait_zapper_service_stable() {
+    previous_addon_count=-1
+    stable_count=0
+    max_stable_checks=3
+    while true; do
+        current_addon_count=$(eval "_run_zapper zapper addon list 2> /dev/null" | wc -l)
+        if [[ "$current_addon_count" -eq "$previous_addon_count" ]]; then
+          ((stable_count++))
+        else
+          stable_count=0
+        fi
+
+        if [[ "$stable_count" -ge "$max_stable_checks" ]]; then
+          break
+        fi
+
+        previous_addon_count="$current_addon_count"
+    done
```

Comment:
Don't you need some sort of sleep here between consecutive polls?

Reply:
no because the `zapper addon list` takes seconds to run, and already gives the right pace